Which is USERNAME and PASSWORD should I put no `DBBACKUP_CONNECTORS`?

I use PostgreSQL 9.5 and use md5 authentication

I set my system User ID and password on `DBBACKUP_CONNECTORS` settings.
```
DBBACKUP_CONNECTORS = {
    'default': {
        'USER': 'rightx2',
        'PASSWORD': 'MY_SYSTEM_PASSWD',
    }
}
```

Whenever I execute command, `python manage.py dbbackup`, it occurred errors:
```
dbbackup.db.exceptions.CommandConnectorError: Error running:  pg_dump super_trading --host=localhost --port=5432 --username=rightx2 --no-password --clean
b'pg_dump: [archiver (db)] connection to database "super_trading" failed: FATAL:  password authentication failed for user "rightx2"\nFATAL:  password authentication failed for user "rightx2"\n'
```

I wonder what I should put on `USER` and `PASSWORD` in  `DBBACKUP_CONNECTORS`

@ZuluPro If I removed `DBBACKUP_CONNECTORS` part in my `settings` and run `python manage.py dbbackup`, same error occurs,
```
dbbackup.db.exceptions.CommandConnectorError: Error running:  pg_dump super_trading --host=localhost --port=5432 --username=rightx2 --no-password --clean
b'pg_dump: [archiver (db)] connection to database "super_trading" failed: FATAL:  password authentication failed for user "rightx2"\nFATAL:  password authentication failed for user "rightx2"\n'
```

So I have checked whether my database's method of authentication is really `md5` .

My default datbase settings:
```
DATABASES['default'] = dj_database_url.config(
    default="postgres://rightx2:MY_SYSTEM_PASSWD!@localhost:5432/super_trading"
)
```

And I tried to change username from `rightx2` to `asdf` and run `python manage.py makemigrations & migrate`, it occurred errors too,
```

    self.connection = self.get_new_connection(conn_params)
  File "/home/rightx2/.pyenv/versions/super_trading_project/lib/python3.6/site-packages/django/db/backends/postgresql/base.py", line 176, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/home/rightx2/.pyenv/versions/super_trading_project/lib/python3.6/site-packages/psycopg2/__init__.py", line 130, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: FATAL:  password authentication failed for user "asdf"
FATAL:  password authentication failed for user "asdf"

Makefile:2: recipe for target 'migrate' failed
make: *** [migrate] Error 1

```
I did the same test for the password, it occurred the same error, too.

According to this, I'm sure that the way of authenticating is `md5`.

But `python manage.py dbbackup` still doesn't work and I don't know why...

--- a/.env-example
+++ b/.env-example
@@ -1,0 +1,8 @@
+# env variables in dbbackup/tests/settings.py
+DB_ENGINE
+DB_NAME
+DB_USER
+DB_PASSWORD
+DB_HOST
+CONNECTOR
+DJANGO_LOG_LEVEL=DEBUG

--- a/.gitignore
+++ b/.gitignore
@@ -7,6 +7,7 @@ __pycache__/
 
 # Distribution / packaging
 .Python
+.env
 env/
 build/
 develop-eggs/
@@ -62,3 +63,4 @@ target/
 .idea/
 *.sw[po]
 test-sqlite
+venv

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,6 @@
 language: python
 
 python:
-  - "3.5"
   - "3.6"
   - "3.7"
   - "3.8"
@@ -70,10 +69,6 @@ matrix:
     - python: "pypy"
       env: DJANGO=3.0
     - python: "pypy"
-      env: DJANGO=master
-    - python: "3.5"
-      env: DJANGO=3.0
-    - python: "3.5"
       env: DJANGO=master
   allow_failures:
     - python: "nightly"

--- a/dbbackup/db/base.py
+++ b/dbbackup/db/base.py
@@ -7,8 +7,14 @@ from django.core.files.base import File
 from tempfile import SpooledTemporaryFile
 from subprocess import Popen
 from importlib import import_module
+import logging
+
 from dbbackup import settings, utils
 from . import exceptions
+
+logger = logging.getLogger('dbbackup.command')
+logger.setLevel(logging.DEBUG)
+
 
 CONNECTOR_MAPPING = {
     'django.db.backends.sqlite3': 'dbbackup.db.sqlite.SqliteConnector',
@@ -128,6 +134,7 @@ class BaseCommandDBConnector(BaseDBConnector):
         :return: Standard output of command
         :rtype: file
         """
+        logger.debug(command)
         cmd = shlex.split(command)
         stdout = SpooledTemporaryFile(max_size=settings.TMP_FILE_MAX_SIZE,
                                       dir=settings.TMP_DIR)

--- a/dbbackup/db/exceptions.py
+++ b/dbbackup/db/exceptions.py
@@ -1,4 +1,6 @@
 """Exceptions for database connectors."""
+
+
 class ConnectorError(Exception):
     """Base connector error"""
 

--- a/dbbackup/db/postgresql.py
+++ b/dbbackup/db/postgresql.py
@@ -1,5 +1,7 @@
-from dbbackup import utils
+from urllib.parse import quote
 import logging
+
+from dbbackup import utils
 from .base import BaseCommandDBConnector
 from .exceptions import DumpError
 
@@ -12,9 +14,9 @@ def create_postgres_uri(self):
         raise DumpError('A host name is required')
 
     dbname = self.settings.get('NAME') or ''
-    user = self.settings.get('USER') or ''
+    user = quote(self.settings.get('USER') or '')
     password = self.settings.get('PASSWORD') or ''
-    password = ':{}'.format(password) if password else ''
+    password = ':{}'.format(quote(password)) if password else ''
     if not user:
         password = ''
     else:

--- a/dbbackup/db/postgresql.py
+++ b/dbbackup/db/postgresql.py
@@ -1,5 +1,19 @@
 from dbbackup import utils
+import logging
 from .base import BaseCommandDBConnector
+
+logger = logging.getLogger('dbbackup.command')
+
+
+def create_postgres_uri(self):
+    host = self.settings.get('HOST', '')
+    dbname = self.settings.get('NAME', '')
+    user = self.settings.get('USER', '')
+    password = self.settings.get('PASSWORD')
+    password = ':{}'.format(password) if password else ''
+    port = ':{}'.format(self.settings.get('PORT')) if self.settings.get('PORT') else ''
+    dbname = f'--dbname=postgresql://{user}{password}@{host}{port}/{dbname}'
+    return dbname
 
 
 class PgDumpConnector(BaseCommandDBConnector):
@@ -22,31 +36,22 @@ class PgDumpConnector(BaseCommandDBConnector):
 
     def _create_dump(self):
         cmd = '{} '.format(self.dump_cmd)
-        if self.settings.get('HOST'):
-            cmd += ' --host={}'.format(self.settings['HOST'])
-        if self.settings.get('PORT'):
-            cmd += ' --port={}'.format(self.settings['PORT'])
-        if self.settings.get('USER'):
-            cmd += ' --username={}'.format(self.settings['USER'])
-        cmd += ' --no-password'
+        cmd = cmd + create_postgres_uri(self)
+
         for table in self.exclude:
             cmd += ' --exclude-table={}'.format(table)
         if self.drop:
             cmd += ' --clean'
-        cmd += ' {}'.format(self.settings['NAME'])
+
         cmd = '{} {} {}'.format(self.dump_prefix, cmd, self.dump_suffix)
+        logger.debug('Postgres cmd: ' + cmd)
         stdout, stderr = self.run_command(cmd, env=self.dump_env)
         return stdout
 
     def _restore_dump(self, dump):
         cmd = '{} '.format(self.restore_cmd)
-        if self.settings.get('HOST'):
-            cmd += ' --host={}'.format(self.settings['HOST'])
-        if self.settings.get('PORT'):
-            cmd += ' --port={}'.format(self.settings['PORT'])
-        if self.settings.get('USER'):
-            cmd += ' --username={}'.format(self.settings['USER'])
-        cmd += ' --no-password'
+        cmd = cmd + create_postgres_uri(self)
+
         # without this, psql terminates with an exit value of 0 regardless of errors
         cmd += ' --set ON_ERROR_STOP=on'
         if self.single_transaction:
@@ -93,14 +98,9 @@ class PgDumpBinaryConnector(PgDumpConnector):
     drop = True
 
     def _create_dump(self):
-        cmd = '{} {}'.format(self.dump_cmd, self.settings['NAME'])
-        if self.settings.get('HOST'):
-            cmd += ' --host={}'.format(self.settings['HOST'])
-        if self.settings.get('PORT'):
-            cmd += ' --port={}'.format(self.settings['PORT'])
-        if self.settings.get('USER'):
-            cmd += ' --user={}'.format(self.settings['USER'])
-        cmd += ' --no-password'
+        cmd = '{} '.format(self.dump_cmd)
+        cmd = cmd + create_postgres_uri(self)
+
         cmd += ' --format=custom'
         for table in self.exclude:
             cmd += ' --exclude-table={}'.format(table)
@@ -109,14 +109,9 @@ class PgDumpBinaryConnector(PgDumpConnector):
         return stdout
 
     def _restore_dump(self, dump):
-        cmd = '{} --dbname={}'.format(self.restore_cmd, self.settings['NAME'])
-        if self.settings.get('HOST'):
-            cmd += ' --host={}'.format(self.settings['HOST'])
-        if self.settings.get('PORT'):
-            cmd += ' --port={}'.format(self.settings['PORT'])
-        if self.settings.get('USER'):
-            cmd += ' --user={}'.format(self.settings['USER'])
-        cmd += ' --no-password'
+        dbname = create_postgres_uri(self)
+        cmd = '{} {}'.format(self.restore_cmd, dbname)
+
         if self.single_transaction:
             cmd += ' --single-transaction'
         if self.drop:

--- a/dbbackup/db/postgresql.py
+++ b/dbbackup/db/postgresql.py
@@ -1,18 +1,27 @@
 from dbbackup import utils
 import logging
 from .base import BaseCommandDBConnector
+from .exceptions import DumpError
 
 logger = logging.getLogger('dbbackup.command')
 
 
 def create_postgres_uri(self):
-    host = self.settings.get('HOST', '')
-    dbname = self.settings.get('NAME', '')
-    user = self.settings.get('USER', '')
-    password = self.settings.get('PASSWORD')
+    host = self.settings.get('HOST')
+    if not host:
+        raise DumpError('A host name is required')
+
+    dbname = self.settings.get('NAME') or ''
+    user = self.settings.get('USER') or ''
+    password = self.settings.get('PASSWORD') or ''
     password = ':{}'.format(password) if password else ''
+    if not user:
+        password = ''
+    else:
+        host = '@' + host
+
     port = ':{}'.format(self.settings.get('PORT')) if self.settings.get('PORT') else ''
-    dbname = f'--dbname=postgresql://{user}{password}@{host}{port}/{dbname}'
+    dbname = f'--dbname=postgresql://{user}{password}{host}{port}/{dbname}'
     return dbname
 
 
@@ -44,7 +53,6 @@ class PgDumpConnector(BaseCommandDBConnector):
             cmd += ' --clean'
 
         cmd = '{} {} {}'.format(self.dump_prefix, cmd, self.dump_suffix)
-        logger.debug('Postgres cmd: ' + cmd)
         stdout, stderr = self.run_command(cmd, env=self.dump_env)
         return stdout
 

--- a/dbbackup/tests/settings.py
+++ b/dbbackup/tests/settings.py
@@ -6,7 +6,7 @@ import tempfile
 import sys
 from dotenv import load_dotenv
 
-test = len(sys.argv) < 0 or sys.argv[1] == 'test'
+test = len(sys.argv) <= 1 or sys.argv[1] == 'test'
 if not test:
     load_dotenv()
 

--- a/dbbackup/tests/settings.py
+++ b/dbbackup/tests/settings.py
@@ -3,8 +3,12 @@ Configuration and launcher for dbbackup tests.
 """
 import os
 import tempfile
+import sys
 from dotenv import load_dotenv
-load_dotenv()
+
+test = len(sys.argv) < 0 or sys.argv[1] == 'test'
+if not test:
+    load_dotenv()
 
 DEBUG = False
 
@@ -26,13 +30,15 @@ INSTALLED_APPS = (
     'dbbackup.tests.testapp',
 )
 
-DATABASES = {'default': {
-    "ENGINE": os.environ.get('DB_ENGINE', "django.db.backends.sqlite3"),
-    "NAME": os.environ.get('DB_NAME', ":memory:"),
-    "USER": os.environ.get('DB_USER'),
-    "PASSWORD": os.environ.get('DB_PASSWORD'),
-    "HOST": os.environ.get('DB_HOST'),
-}}
+DATABASES = {
+    'default': {
+        "ENGINE": os.environ.get('DB_ENGINE', "django.db.backends.sqlite3"),
+        "NAME": os.environ.get('DB_NAME', ":memory:"),
+        "USER": os.environ.get('DB_USER'),
+        "PASSWORD": os.environ.get('DB_PASSWORD'),
+        "HOST": os.environ.get('DB_HOST'),
+    }
+}
 if os.environ.get('CONNECTOR'):
     CONNECTOR = {'CONNECTOR': os.environ['CONNECTOR']}
     DBBACKUP_CONNECTORS = {'default': CONNECTOR}

--- a/dbbackup/tests/settings.py
+++ b/dbbackup/tests/settings.py
@@ -3,6 +3,8 @@ Configuration and launcher for dbbackup tests.
 """
 import os
 import tempfile
+from dotenv import load_dotenv
+load_dotenv()
 
 DEBUG = False
 
@@ -50,3 +52,35 @@ DBBACKUP_STORAGE = os.environ.get('STORAGE', 'dbbackup.tests.utils.FakeStorage')
 DBBACKUP_STORAGE_OPTIONS = dict([keyvalue.split('=') for keyvalue in
                                  os.environ.get('STORAGE_OPTIONS', '').split(',')
                                  if keyvalue])
+
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,
+    'root': {
+        'handlers': ['console'],
+        'level': 'DEBUG'
+    },
+    'handlers': {
+        'console': {
+            'level': os.getenv('DJANGO_LOG_LEVEL', 'INFO'),
+            'class': 'logging.StreamHandler',
+            'formatter': 'simple'
+        }
+    },
+    'formatters': {
+        'verbose': {
+            'format': "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
+            'datefmt': "%d/%b/%Y %H:%M:%S"
+        },
+        'simple': {
+            'format': '%(levelname)s %(message)s'
+        },
+    },
+    'loggers': {
+        'django.db.backends': {
+            # uncomment to see all queries
+            # 'level': 'DEBUG',
+            'handlers': ['console'],
+        }
+    }
+}

--- a/dbbackup/tests/test_connectors/test_postgresql.py
+++ b/dbbackup/tests/test_connectors/test_postgresql.py
@@ -21,6 +21,14 @@ class PgDumpConnectorTest(TestCase):
         self.connector.settings['NAME'] = 'dbname'
         self.connector.settings['HOST'] = 'hostname'
 
+    def test_user_password_uses_special_characters(self, mock_dump_cmd):
+        self.connector.settings['PASSWORD'] = '@!'
+        self.connector.settings['USER'] = '@'
+
+        self.connector.create_dump()
+
+        self.assertIn('postgresql://%40:%40%21@hostname/dbname', mock_dump_cmd.call_args[0][0])
+
     def test_create_dump(self, mock_dump_cmd):
         dump = self.connector.create_dump()
         # Test dump

--- a/dbbackup/tests/test_connectors/test_postgresql.py
+++ b/dbbackup/tests/test_connectors/test_postgresql.py
@@ -1,20 +1,28 @@
 from __future__ import unicode_literals
 
-from mock import patch, mock_open
-
+from mock import patch
 from django.test import TestCase
 from six import BytesIO
 
-from dbbackup.db.postgresql import (PgDumpConnector, PgDumpGisConnector,
-                                    PgDumpBinaryConnector)
+from dbbackup.db.postgresql import (
+    PgDumpConnector,
+    PgDumpGisConnector,
+    PgDumpBinaryConnector
+)
+from dbbackup.db.exceptions import DumpError
 
 
 @patch('dbbackup.db.postgresql.PgDumpConnector.run_command',
        return_value=(BytesIO(b'foo'), BytesIO()))
 class PgDumpConnectorTest(TestCase):
+    def setUp(self):
+        self.connector = PgDumpConnector()
+        self.connector.settings['ENGINE'] = 'django.db.backends.postgresql'
+        self.connector.settings['NAME'] = 'dbname'
+        self.connector.settings['HOST'] = 'hostname'
+
     def test_create_dump(self, mock_dump_cmd):
-        connector = PgDumpConnector()
-        dump = connector.create_dump()
+        dump = self.connector.create_dump()
         # Test dump
         dump_content = dump.read()
         self.assertTrue(dump_content)
@@ -22,287 +30,196 @@ class PgDumpConnectorTest(TestCase):
         # Test cmd
         self.assertTrue(mock_dump_cmd.called)
 
+    def test_create_dump_without_host_raises_error(self, mock_dump_cmd):
+        self.connector.settings.pop('HOST', None)
+        with self.assertRaises(DumpError):
+            self.connector.create_dump()
+
+    def test_password_but_no_user(self, mock_dump_cmd):
+        self.connector.settings.pop('USER', None)
+        self.connector.settings['PASSWORD'] = 'hello'
+        self.connector.create_dump()
+        self.assertIn('postgresql://hostname/dbname', mock_dump_cmd.call_args[0][0])
+
     def test_create_dump_host(self, mock_dump_cmd):
-        connector = PgDumpConnector()
-        # Without
-        connector.settings.pop('HOST', None)
-        connector.create_dump()
-        self.assertNotIn(' --host=', mock_dump_cmd.call_args[0][0])
         # With
-        connector.settings['HOST'] = 'foo'
-        connector.create_dump()
-        self.assertIn(' --host=foo', mock_dump_cmd.call_args[0][0])
+        self.connector.settings['HOST'] = 'foo'
+        self.connector.create_dump()
+        self.assertIn('postgresql://foo/dbname', mock_dump_cmd.call_args[0][0])
 
     def test_create_dump_port(self, mock_dump_cmd):
-        connector = PgDumpConnector()
         # Without
-        connector.settings.pop('PORT', None)
-        connector.create_dump()
-        self.assertNotIn(' --port=', mock_dump_cmd.call_args[0][0])
+        self.connector.settings.pop('PORT', None)
+        self.connector.create_dump()
+        self.assertIn('postgresql://hostname/dbname', mock_dump_cmd.call_args[0][0])
         # With
-        connector.settings['PORT'] = 42
-        connector.create_dump()
-        self.assertIn(' --port=42', mock_dump_cmd.call_args[0][0])
+        self.connector.settings['PORT'] = 42
+        self.connector.create_dump()
+        self.assertIn('postgresql://hostname:42/dbname', mock_dump_cmd.call_args[0][0])
 
     def test_create_dump_user(self, mock_dump_cmd):
-        connector = PgDumpConnector()
         # Without
-        connector.settings.pop('USER', None)
-        connector.create_dump()
-        self.assertNotIn(' --username=', mock_dump_cmd.call_args[0][0])
+        self.connector.settings.pop('USER', None)
+        self.connector.create_dump()
+        self.assertIn('postgresql://hostname/dbname', mock_dump_cmd.call_args[0][0])
         # With
-        connector.settings['USER'] = 'foo'
-        connector.create_dump()
-        self.assertIn(' --username=foo', mock_dump_cmd.call_args[0][0])
+        self.connector.settings['USER'] = 'foo'
+        self.connector.create_dump()
+        self.assertIn('postgresql://foo@hostname/dbname', mock_dump_cmd.call_args[0][0])
 
     def test_create_dump_exclude(self, mock_dump_cmd):
-        connector = PgDumpConnector()
         # Without
-        connector.create_dump()
+        self.connector.create_dump()
         self.assertNotIn(' --exclude-table=', mock_dump_cmd.call_args[0][0])
         # With
-        connector.exclude = ('foo',)
-        connector.create_dump()
+        self.connector.exclude = ('foo',)
+        self.connector.create_dump()
         self.assertIn(' --exclude-table=foo', mock_dump_cmd.call_args[0][0])
         # With serveral
-        connector.exclude = ('foo', 'bar')
-        connector.create_dump()
+        self.connector.exclude = ('foo', 'bar')
+        self.connector.create_dump()
         self.assertIn(' --exclude-table=foo', mock_dump_cmd.call_args[0][0])
         self.assertIn(' --exclude-table=bar', mock_dump_cmd.call_args[0][0])
 
     def test_create_dump_drop(self, mock_dump_cmd):
-        connector = PgDumpConnector()
         # Without
-        connector.drop = False
-        connector.create_dump()
+        self.connector.drop = False
+        self.connector.create_dump()
         self.assertNotIn(' --clean', mock_dump_cmd.call_args[0][0])
         # With
-        connector.drop = True
-        connector.create_dump()
+        self.connector.drop = True
+        self.connector.create_dump()
         self.assertIn(' --clean', mock_dump_cmd.call_args[0][0])
 
     @patch('dbbackup.db.postgresql.PgDumpConnector.run_command',
            return_value=(BytesIO(), BytesIO()))
     def test_restore_dump(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpConnector()
-        dump = connector.create_dump()
-        connector.restore_dump(dump)
+        dump = self.connector.create_dump()
+        self.connector.restore_dump(dump)
         # Test cmd
         self.assertTrue(mock_restore_cmd.called)
 
     @patch('dbbackup.db.postgresql.PgDumpConnector.run_command',
            return_value=(BytesIO(), BytesIO()))
-    def test_restore_dump_host(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpConnector()
-        dump = connector.create_dump()
-        # Without
-        connector.settings.pop('HOST', None)
-        connector.restore_dump(dump)
-        self.assertNotIn(' --host=foo', mock_restore_cmd.call_args[0][0])
-        # With
-        connector.settings['HOST'] = 'foo'
-        connector.restore_dump(dump)
-        self.assertIn(' --host=foo', mock_restore_cmd.call_args[0][0])
-
-    @patch('dbbackup.db.postgresql.PgDumpConnector.run_command',
-           return_value=(BytesIO(), BytesIO()))
-    def test_restore_dump_port(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpConnector()
-        dump = connector.create_dump()
-        # Without
-        connector.settings.pop('PORT', None)
-        connector.restore_dump(dump)
-        self.assertNotIn(' --port=', mock_restore_cmd.call_args[0][0])
-        # With
-        connector.settings['PORT'] = 42
-        connector.restore_dump(dump)
-        self.assertIn(' --port=42', mock_restore_cmd.call_args[0][0])
-
-    @patch('dbbackup.db.postgresql.PgDumpConnector.run_command',
-           return_value=(BytesIO(), BytesIO()))
     def test_restore_dump_user(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpConnector()
-        dump = connector.create_dump()
+        dump = self.connector.create_dump()
         # Without
-        connector.settings.pop('USER', None)
-        connector.restore_dump(dump)
+        self.connector.settings.pop('USER', None)
+        self.connector.restore_dump(dump)
+
+        self.assertIn(
+            'postgresql://hostname/dbname',
+            mock_restore_cmd.call_args[0][0]
+        )
+
         self.assertNotIn(' --username=', mock_restore_cmd.call_args[0][0])
         # With
-        connector.settings['USER'] = 'foo'
-        connector.restore_dump(dump)
-        self.assertIn(' --username=foo', mock_restore_cmd.call_args[0][0])
+        self.connector.settings['USER'] = 'foo'
+        self.connector.restore_dump(dump)
+        self.assertIn(
+            'postgresql://foo@hostname/dbname',
+            mock_restore_cmd.call_args[0][0]
+        )
 
 
 @patch('dbbackup.db.postgresql.PgDumpBinaryConnector.run_command',
        return_value=(BytesIO(b'foo'), BytesIO()))
 class PgDumpBinaryConnectorTest(TestCase):
+    def setUp(self):
+        self.connector = PgDumpBinaryConnector()
+        self.connector.settings['HOST'] = 'hostname'
+        self.connector.settings['ENGINE'] = 'django.db.backends.postgresql'
+        self.connector.settings['NAME'] = 'dbname'
+
     def test_create_dump(self, mock_dump_cmd):
-        connector = PgDumpBinaryConnector()
-        dump = connector.create_dump()
+        dump = self.connector.create_dump()
         # Test dump
         dump_content = dump.read()
         self.assertTrue(dump_content)
         self.assertEqual(dump_content, b'foo')
         # Test cmd
         self.assertTrue(mock_dump_cmd.called)
-
-    def test_create_dump_host(self, mock_dump_cmd):
-        connector = PgDumpBinaryConnector()
-        # Without
-        connector.settings.pop('HOST', None)
-        connector.create_dump()
-        self.assertNotIn(' --host=', mock_dump_cmd.call_args[0][0])
-        # With
-        connector.settings['HOST'] = 'foo'
-        connector.create_dump()
-        self.assertIn(' --host=foo', mock_dump_cmd.call_args[0][0])
-
-    def test_create_dump_port(self, mock_dump_cmd):
-        connector = PgDumpBinaryConnector()
-        # Without
-        connector.settings.pop('PORT', None)
-        connector.create_dump()
-        self.assertNotIn(' --port=', mock_dump_cmd.call_args[0][0])
-        # With
-        connector.settings['PORT'] = 42
-        connector.create_dump()
-        self.assertIn(' --port=42', mock_dump_cmd.call_args[0][0])
-
-    def test_create_dump_user(self, mock_dump_cmd):
-        connector = PgDumpBinaryConnector()
-        # Without
-        connector.settings.pop('USER', None)
-        connector.create_dump()
-        self.assertNotIn(' --user=', mock_dump_cmd.call_args[0][0])
-        # With
-        connector.settings['USER'] = 'foo'
-        connector.create_dump()
-        self.assertIn(' --user=foo', mock_dump_cmd.call_args[0][0])
+        self.assertIn('--format=custom', mock_dump_cmd.call_args[0][0])
 
     def test_create_dump_exclude(self, mock_dump_cmd):
-        connector = PgDumpBinaryConnector()
         # Without
-        connector.create_dump()
+        self.connector.create_dump()
         self.assertNotIn(' --exclude-table=', mock_dump_cmd.call_args[0][0])
         # With
-        connector.exclude = ('foo',)
-        connector.create_dump()
+        self.connector.exclude = ('foo',)
+        self.connector.create_dump()
         self.assertIn(' --exclude-table=foo', mock_dump_cmd.call_args[0][0])
         # With serveral
-        connector.exclude = ('foo', 'bar')
-        connector.create_dump()
+        self.connector.exclude = ('foo', 'bar')
+        self.connector.create_dump()
         self.assertIn(' --exclude-table=foo', mock_dump_cmd.call_args[0][0])
         self.assertIn(' --exclude-table=bar', mock_dump_cmd.call_args[0][0])
 
     def test_create_dump_drop(self, mock_dump_cmd):
-        connector = PgDumpBinaryConnector()
         # Without
-        connector.drop = False
-        connector.create_dump()
+        self.connector.drop = False
+        self.connector.create_dump()
         self.assertNotIn(' --clean', mock_dump_cmd.call_args[0][0])
         # Binary drop at restore level
-        connector.drop = True
-        connector.create_dump()
+        self.connector.drop = True
+        self.connector.create_dump()
         self.assertNotIn(' --clean', mock_dump_cmd.call_args[0][0])
 
     @patch('dbbackup.db.postgresql.PgDumpBinaryConnector.run_command',
            return_value=(BytesIO(), BytesIO()))
     def test_restore_dump(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpBinaryConnector()
-        dump = connector.create_dump()
-        connector.restore_dump(dump)
+        dump = self.connector.create_dump()
+        self.connector.restore_dump(dump)
         # Test cmd
         self.assertTrue(mock_restore_cmd.called)
-
-    @patch('dbbackup.db.postgresql.PgDumpBinaryConnector.run_command',
-           return_value=(BytesIO(), BytesIO()))
-    def test_restore_dump_host(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpBinaryConnector()
-        dump = connector.create_dump()
-        # Without
-        connector.settings.pop('HOST', None)
-        connector.restore_dump(dump)
-        self.assertNotIn(' --host=foo', mock_restore_cmd.call_args[0][0])
-        # With
-        connector.settings['HOST'] = 'foo'
-        connector.restore_dump(dump)
-        self.assertIn(' --host=foo', mock_restore_cmd.call_args[0][0])
-
-    @patch('dbbackup.db.postgresql.PgDumpBinaryConnector.run_command',
-           return_value=(BytesIO(), BytesIO()))
-    def test_restore_dump_port(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpBinaryConnector()
-        dump = connector.create_dump()
-        # Without
-        connector.settings.pop('PORT', None)
-        connector.restore_dump(dump)
-        self.assertNotIn(' --port=', mock_restore_cmd.call_args[0][0])
-        # With
-        connector.settings['PORT'] = 42
-        connector.restore_dump(dump)
-        self.assertIn(' --port=42', mock_restore_cmd.call_args[0][0])
-
-    @patch('dbbackup.db.postgresql.PgDumpBinaryConnector.run_command',
-           return_value=(BytesIO(), BytesIO()))
-    def test_restore_dump_user(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpBinaryConnector()
-        dump = connector.create_dump()
-        # Without
-        connector.settings.pop('USER', None)
-        connector.restore_dump(dump)
-        self.assertNotIn(' --user=', mock_restore_cmd.call_args[0][0])
-        # With
-        connector.settings['USER'] = 'foo'
-        connector.restore_dump(dump)
-        self.assertIn(' --user=foo', mock_restore_cmd.call_args[0][0])
 
 
 @patch('dbbackup.db.postgresql.PgDumpGisConnector.run_command',
        return_value=(BytesIO(b'foo'), BytesIO()))
 class PgDumpGisConnectorTest(TestCase):
+    def setUp(self):
+        self.connector = PgDumpGisConnector()
+        self.connector.settings['HOST'] = 'hostname'
+
     @patch('dbbackup.db.postgresql.PgDumpGisConnector.run_command',
            return_value=(BytesIO(b'foo'), BytesIO()))
     def test_restore_dump(self, mock_dump_cmd, mock_restore_cmd):
-        connector = PgDumpGisConnector()
-        dump = connector.create_dump()
+        dump = self.connector.create_dump()
         # Without ADMINUSER
-        connector.settings.pop('ADMIN_USER', None)
-        connector.restore_dump(dump)
+        self.connector.settings.pop('ADMIN_USER', None)
+        self.connector.restore_dump(dump)
         self.assertTrue(mock_restore_cmd.called)
         # With
-        connector.settings['ADMIN_USER'] = 'foo'
-        connector.restore_dump(dump)
+        self.connector.settings['ADMIN_USER'] = 'foo'
+        self.connector.restore_dump(dump)
         self.assertTrue(mock_restore_cmd.called)
 
     def test_enable_postgis(self, mock_dump_cmd):
-        connector = PgDumpGisConnector()
-        connector.settings['ADMIN_USER'] = 'foo'
-        connector._enable_postgis()
+        self.connector.settings['ADMIN_USER'] = 'foo'
+        self.connector._enable_postgis()
         self.assertIn('"CREATE EXTENSION IF NOT EXISTS postgis;"', mock_dump_cmd.call_args[0][0])
         self.assertIn('--username=foo', mock_dump_cmd.call_args[0][0])
 
     def test_enable_postgis_host(self, mock_dump_cmd):
-        connector = PgDumpGisConnector()
-        connector.settings['ADMIN_USER'] = 'foo'
+        self.connector.settings['ADMIN_USER'] = 'foo'
         # Without
-        connector.settings.pop('HOST', None)
-        connector._enable_postgis()
+        self.connector.settings.pop('HOST', None)
+        self.connector._enable_postgis()
         self.assertNotIn(' --host=', mock_dump_cmd.call_args[0][0])
         # With
-        connector.settings['HOST'] = 'foo'
-        connector._enable_postgis()
+        self.connector.settings['HOST'] = 'foo'
+        self.connector._enable_postgis()
         self.assertIn(' --host=foo', mock_dump_cmd.call_args[0][0])
 
     def test_enable_postgis_port(self, mock_dump_cmd):
-        connector = PgDumpGisConnector()
-        connector.settings['ADMIN_USER'] = 'foo'
+        self.connector.settings['ADMIN_USER'] = 'foo'
         # Without
-        connector.settings.pop('PORT', None)
-        connector._enable_postgis()
+        self.connector.settings.pop('PORT', None)
+        self.connector._enable_postgis()
         self.assertNotIn(' --port=', mock_dump_cmd.call_args[0][0])
         # With
-        connector.settings['PORT'] = 42
-        connector._enable_postgis()
+        self.connector.settings['PORT'] = 42
+        self.connector._enable_postgis()
         self.assertIn(' --port=42', mock_dump_cmd.call_args[0][0])
 
 
@@ -311,13 +228,16 @@ class PgDumpGisConnectorTest(TestCase):
     'return_value.poll.return_value': False,
 })
 class PgDumpConnectorRunCommandTest(TestCase):
+
     def test_run_command(self, mock_popen):
         connector = PgDumpConnector()
+        connector.settings['HOST'] = 'hostname'
         connector.create_dump()
         self.assertEqual(mock_popen.call_args[0][0][0], 'pg_dump')
 
     def test_run_command_with_password(self, mock_popen):
         connector = PgDumpConnector()
+        connector.settings['HOST'] = 'hostname'
         connector.settings['PASSWORD'] = 'foo'
         connector.create_dump()
         self.assertEqual(mock_popen.call_args[0][0][0], 'pg_dump')
@@ -326,6 +246,7 @@ class PgDumpConnectorRunCommandTest(TestCase):
 
     def test_run_command_with_password_and_other(self, mock_popen):
         connector = PgDumpConnector(env={'foo': 'bar'})
+        connector.settings['HOST'] = 'hostname'
         connector.settings['PASSWORD'] = 'foo'
         connector.create_dump()
         self.assertEqual(mock_popen.call_args[0][0][0], 'pg_dump')

--- a/requirements-docs.txt
+++ b/requirements-docs.txt
@@ -1,3 +1,4 @@
 Sphinx
 docutils
 sphinx-django-command
+python-dotenv

--- a/requirements-tests.txt
+++ b/requirements-tests.txt
@@ -7,3 +7,5 @@ django-storages
 pytz
 testfixtures
 mock
+python-dotenv
+psycopg2

--- a/tox.ini
+++ b/tox.ini
@@ -1,12 +1,11 @@
 [tox]
-envlist = py{3.5,3.6,3.7,3.8,nightly}-django2.2,py{3.6,3.7,3.8,nightly}-django{3.0,master},lint,docs,functional
+envlist = py{3.6,3.7,3.8,nightly}-django2.2,py{3.6,3.7,3.8,nightly}-django{3.0,master},lint,docs,functional
 
 [testenv]
 passenv = *
 setenv =
     PYTHONDONTWRITEBYTECODE=1
 basepython =
-    py3.5: python3.5
     py3.6: python3.6
     py3.7: python3.7
     py3.8: python3.8

